### ISSUE TITLE
Issue resolving jmods dir when cross linking from linux to a windows JDK

### ISSUE THREAD

**Issue by recursinging** [moditect/moditect#63]
There is a little problem resolving the jmods directory of the target JDK when attempting to cross-link from linux to a windows JDK.  The windows openjdk build names javac as "javac.exe" which does not get resolved by this routine and throws a NPE. 

https://github.com/moditect/moditect/blob/d4633e42fa5e731c72eb8c50aee9fbf21ced77d8/maven-plugin/src/main/java/org/moditect/mavenplugin/image/CreateRuntimeImageMojo.java#L133

I worked around it by making a symlink from "javac.exe" -> "javac". Everything worked as expected afterward and I got a working win64 binary linked together on a linux box (The future has arrived!). 

Sorry for the lack of a patch, too little time for all that, so just this heads up.  Thanks for this awesome project!

**Comment by gunnarmorling** [comment]
Thanks for the report; would you happen to know whether it worked if we were to use `findTool( "javac.exe" )` when running on Windows? I can add this, but I have no way of testing this on Windows unfortunately.

**Comment by recursinging** [comment]
I think so, the maven [Toolchain Interface](http://maven.apache.org/ref/3-LATEST/maven-core/apidocs/org/apache/maven/toolchain/Toolchain.html#findTool(java.lang.String)) documents the `findTool` argument as as  _"the tool platform independent tool name."_

Looking at [the code](https://github.com/apache/maven/blob/master/maven-core/src/main/java/org/apache/maven/toolchain/java/JavaToolchainImpl.java), I think calling `findTool( "javac.exe" )` would work as long as it is guarded by a null check of `findTool("javac")`, for example.:  

```
File javac = toolChains.get( 0 ).findTool( "javac" );
if (javac == null) {
    javac = toolChains.get( 0 ).findTool( "javac.exe" );
}
return javac.toPath()
    .getParent()
    .getParent()
     .resolve( "jmods" );
```
...that way it should still work on windows, with a windows JDK, and should even work with a linux JDK on windows, should one want to cross compile from there.

**Comment by gunnarmorling** [comment]
Hum, in the code you linked, it adds the ".exe" suffix automatically: https://github.com/apache/maven/blob/master/maven-core/src/main/java/org/apache/maven/toolchain/java/JavaToolchainImpl.java#L79. It seems that logic hasn't triggered in your case for some reason.

**Comment by recursinging** [comment]
The .exe extension is added if the executing platform is Windows. Invoking this plugin in maven on Linux (while cross compiling  for a Windows target) does not trigger this behavior. 

It's confusing but logical. The toolchain is being resolved by the build platform, not the target platform.  

The null check I mentioned would protect against a role reversal, a Windows build targeting Linux.

**Comment by gunnarmorling** [comment]
Ah yes, gotcha. It's caused by the fact that we sort of misuse `findTool()` there to locate the basedir of that tool chain entry. I'll add the logic to try `javac.exe`, too.

**Comment by gunnarmorling** [comment]
Fixed via #74.
